src/main/java/seedu/ultron/logic/parser/UpcomingCommandParser.java

```diff
@@ -0,0 +1,28 @@
+package seedu.ultron.logic.parser;
+
+import static seedu.ultron.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.ultron.logic.commands.UpcomingCommand;
+import seedu.ultron.logic.parser.exceptions.ParseException;
+import seedu.ultron.model.opening.OpeningsBeforeDaysPredicate;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class UpcomingCommandParser implements Parser<UpcomingCommand> {
+    public UpcomingCommand parse(String args) throws ParseException {
+        try {
+            String trimmedArgs = args.trim();
+            int days = ParserUtil.parseDays(trimmedArgs);
+            return new UpcomingCommand(new OpeningsBeforeDaysPredicate(days));
+        } catch (ParseException pe) {
```

Comment:
No need to catch the ParseException, the parse method already throws the exception.

Reply:
When implementing parseDays, I referred to the parseIndex function which also throws exception for unsigned non-zero integers.